Tests/SonarScanner.MSBuild.Shim.Test/PathHelperTests.cs

```diff
@@ -105,84 +106,126 @@ public void IsInDirectory_ReturnsTheExpectedResults()
         }
 
         [TestMethod]
-        public void BestCommonPrefix_WhenNull_ReturnsNull() =>
-            PathHelper.BestCommonPrefix(null).Should().BeNull();
+        public void BestCommonPrefix_WhenParametersAreNull_ReturnsNull()
+        {
+            PathHelper.BestCommonPrefix(null, StringComparer.Ordinal).Should().BeNull();
+            PathHelper.BestCommonPrefix(new DirectoryInfo[] {}, null).Should().BeNull();
+            PathHelper.BestCommonPrefix(null, null).Should().BeNull();
+        }
 
         [TestMethod]
         public void BestCommonPrefix_WhenEmpty_ReturnsNull() =>
-            PathHelper.BestCommonPrefix(Enumerable.Empty<DirectoryInfo>()).Should().BeNull();
+            PathHelper.BestCommonPrefix([], StringComparer.Ordinal).Should().BeNull();
 
         [TestMethod]
         public void BestCommonPrefix_WhenNoCommonPath_ReturnsNull_Simple() =>
-            PathHelper.BestCommonPrefix(new[]
-            {
-                new DirectoryInfo(@"C:\"),
-                new DirectoryInfo(@"D:\Dir"),
-            }).Should().BeNull();
+            PathHelper.BestCommonPrefix(
+                new[]
```

Comment:
nitpick: All of these can use collection expressions, like so:

![image](https://github.com/user-attachments/assets/af26aca8-581f-4c7a-bd45-4301bde8d9ca)


It's a bit boring to fix them, so consider this an optional comment.